### PR TITLE
Ft user profile

### PR DESCRIPTION
**Pull request by manaraph**
## Description
User View Profile

All pull requests should be related to open issues. Indicate the issue(s) bellow and remove this line:
Fixes #(issue number 1) #(issue number 2 if applicable)

## How Has This Been Tested?
Please describe in detail how you tested your changes.
Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code etc.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [x] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/controller/userController.js [hunk 1]
```diff
@@ -288,3 +288,66 @@ export const userSignin = (req, res, next) => {
       });
     });
 };
+
+/**
+ *
+ * @api {post} /api/v1/users/:userId Login an existing user
```
Comment (MicahBala) [inline src/controller/userController.js:294]: The route as implemented is a GET route " @api {get} /api/v1/users/userId"

File: src/controller/userController.js [hunk 2]
```diff
@@ -288,3 +288,66 @@ export const userSignin = (req, res, next) => {
...
+ *      error: "Error information"
+ *    }
+ * }
+ */
+
+export const getUserProfile = async (req, res, next) => {
+  const { userId } = req.params;
+  try {
+    // Find the user by primary key.
+    const user = await User.findByPk(userId);
```
Comment (MicahBala) [inline src/controller/userController.js:336]: A none existing userId in the database still returns a 200 O.K "User found" with user data null.  An "If" check could be performed and return error "user not found if userId doesn't exist.

Comment (manaraph) [reply src/controller/userController.js:336]: Thank you, have updated this.

File: src/controller/userController.js (hunk 1 above)
Comment (manaraph) [reply src/controller/userController.js]: Updated, thanks.

File: __tests__/user.test.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    server = http.createServer(app);
+    server.listen(done);
+    request = supertest(server);
+    jest.setTimeout(10 * 1000);
+  });
+
+  afterAll((done) => {
+    server.close(done);
+  });
+
```
Comment (wincodes) [inline __tests__/user.test.js:24]: add a test for unauthenticated user
write test for user not found error

File: __tests__/user.test.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      password: 'dhye%$&6*'
+    };
+
+    const loginResponse = await request.post('/api/v1/auth/signin').send(loginDetails);
+
+    const res = await request.get(`/api/v1/users/${registerResponse.body.data.userId}`).set('Authorization', `Bearer ${loginResponse.body.data.token}`);
+
+    expect(res.statusCode).toEqual(200);
+    expect(res.body.status).toBe('success');
+    expect(res.body.data.message).toBe('User found');
```
Comment (wincodes) [inline __tests__/user.test.js:52]: verify the user Id
also ensure the user password is not returned

File: src/controller/userController.js [hunk 5]
```diff
@@ -288,3 +288,76 @@ export const userSignin = (req, res, next) => {
...
+
+    if (user) {
+      return res.status(200).json({
+        status: 'success',
+        data: {
+          message: 'User found',
+          user
```
Comment (wincodes) [inline src/controller/userController.js:343]: return the user.dataValues only

File: src/controller/userController.js [hunk 6]
```diff
@@ -288,3 +288,76 @@ export const userSignin = (req, res, next) => {
...
+        }
+      });
+    }
+    return res.status(404).json({
+      status: 'error',
+      data: {
+        message: 'User does not exist.',
+        user
```
Comment (wincodes) [inline src/controller/userController.js:351]: user is not required here

File: src/controller/userController.js [hunk 7]
```diff
@@ -288,3 +288,76 @@ export const userSignin = (req, res, next) => {
...
+        status: 'success',
+        data: {
+          message: 'User found',
+          user
+        }
+      });
+    }
+    return res.status(404).json({
+      status: 'error',
+      data: {
```
Comment (wincodes) [inline src/controller/userController.js:349]: let this be errors: { message:

File: src/controller/userController.js [hunk 8]
```diff
@@ -288,3 +288,76 @@ export const userSignin = (req, res, next) => {
...
+        message: 'User does not exist.',
+        user
+      }
+    });
+  } catch (error) {
+    return res.status(500).json({
+      status: 'error',
+      data: {
```
Comment (wincodes) [inline src/controller/userController.js:357]: let this be errors { message:

**Pull request merged**